common/src/main/java/com/emc/mongoose/common/concurrent/DaemonBase.java

```diff
@@ -33,6 +34,9 @@ protected abstract void doClose()
 	public final void start()
 	throws IllegalStateException {
 		if(stateRef.compareAndSet(State.INITIAL, State.STARTED)) {
+			synchronized(state) {
```

Comment:
in order to call notifyAll one should enter the synchronization block on the object used as monitor